### PR TITLE
Modify linkToUrl() to use display_url and expanded_url in the input entity.

### PR DESCRIPTION
**Pull request by keitaf**
This modifies linkToUrl() to use display_url and expanded_url if they are available in the input entity.

This still uses options.urlEntities if provided.

### CODE REVIEW

File: twitter-text.js [hunk 1]
```diff
@@ -519,6 +459,73 @@ if (typeof twttr === "undefined" || twttr === null) {
...
+      //   nname.com/foo
+      // </span>
+      // <span class='tco-ellipsis'> <!-- This stuff should get displayed but not copied -->
+      //   <span style='font-size:0'>&nbsp;</span>
+      //   …
+      // </span>
+      v['invisible'] = "style='font-size:0; line-height:0'";
+      return stringSupplant("<span class='tco-ellipsis'>#{precedingEllipsis}<span #{invisible}>&nbsp;</span></span><span #{invisible}>#{beforeDisplayUrl}</span><span class='js-display-url'>#{displayUrlSansEllipses}</span><span #{invisible}>#{afterDisplayUrl}</span><span class='tco-ellipsis'><span #{invisible}>&nbsp;</span>#{followingEllipsis}</span>", v);
+    }
+    return dispalyUrl;
```
Comment (couch) [inline twitter-text.js]: `displayUrl` is misspelled here.

File: twitter-text.js [hunk 2]
```diff
@@ -519,6 +459,73 @@ if (typeof twttr === "undefined" || twttr === null) {
     return stringSupplant("<a href=\"#{url}\"#{htmlAttrs}>#{linkText}</a>", d);
   };
 
+  twttr.txt.linkTextWithEntity = function (entity) {
```
Comment (couch) [inline twitter-text.js]: This function should probably have a test. It's not entirely clear what form the `entity` object should take.

File: twitter-text.js (hunk 1 above)
Comment (keitaf) [inline twitter-text.js]: fixed.

File: twitter-text.js (hunk 2 above)
Comment (keitaf) [inline twitter-text.js]: Thank you for the review! I added unit tests for `linkTextWithEntity` and also fixed a typo.

**Comment by hoverbird** [comment]
Awesome, tests and code look good to me.

**Pull request merged**